WcaOnRails/lib/time_limit.rb

```diff
@@ -1,16 +1,29 @@
 # frozen_string_literal: true
 
 class TimeLimit
+  include ActiveModel::Validations
+
   TEN_MINUTES_IN_CENTISECONDS = 10*60*100
 
-  attr_accessor :centiseconds, :cumulative_round_ids
-  def initialize(centiseconds: TEN_MINUTES_IN_CENTISECONDS, cumulative_round_ids: [].freeze)
+  attr_accessor :centiseconds, :cumulativeRoundIds
+  validates :centiseconds, numericality: { only_integer: true }
+  validate do
+    unless self.cumulativeRoundIds.is_a?(Array) && self.cumulativeRoundIds.all? { |el| el.is_a?(String) }
+      errors.add("cumulativeRoundIds must be an Array of Strings")
+    end
+  end
+
+  def initialize(centiseconds: TEN_MINUTES_IN_CENTISECONDS, cumulativeRoundIds: [].freeze)
     self.centiseconds = centiseconds
-    self.cumulative_round_ids = cumulative_round_ids
+    self.cumulativeRoundIds = cumulativeRoundIds
+  end
+
+  def cumulativeRoundIds=(cumulativeRoundIds) # rubocop:disable Style/MethodName
```

Comment:
Having a setter function on an `attr_accessor` attribute is quite weird, and I'm surprised the language allows it.

It looks like it's to change nils to `[]`, but there is a validation against nil values, and you can add a `|| []` to line 41.